skyportal/utils/offset_util.py

```diff
@@ -156,26 +313,374 @@ def get_nearby_offset_stars(source_ra, source_dec, source_name,
+    wcs.wcs.ctype = ["RA---TAN", "DEC--TAN"]
+
+    if hdu is not None:
+        im = hdu.data
+
+        # replace the nans with medians
+        im[np.isnan(im)] = np.nanmedian(im)
+        if source_image_parameters[image_source].get("reproject", False):
+            # project image to the skeleton WCS solution
+            print("Reprojecting image to requested position and orientation")
+            im, _ = reproject_adaptive(hdu, wcs, shape_out=(npixels,npixels))
+        else:
+            wcs = WCS(hdu.header)
+
+        if source_image_parameters[image_source].get("smooth", False):
+            im = gaussian_filter(hdu.data,
+                                 source_image_parameters[image_source]["smooth"]/pixscale)
+
+        norm = ImageNormalize(im, interval=ZScaleInterval())
+        watermark = source_image_parameters[image_source]["str"]
+
+    else:
+        # if we got back a blank image, try to fallback on another survey
+        # and return the results from that call
+        if (fallback_image_source is not None):
+            if (fallback_image_source != image_source):
+                print(f"Falling back on image source {fallback_image_source}")
+                return get_finding_chart(source_ra, source_dec, source_name,
+                                         image_source=fallback_image_source,
+                                         output_format=output_format,
+                                         imsize=imsize,
+                                         tick_offset=tick_offset,
+                                         tick_length=tick_length,
+                                         fallback_image_source=None,
+                                         **offset_star_kwargs)
+
+        # we dont have an image here, so let's create a dummy one
+        # so we can still plot
+        im = np.zeros((npixels, npixels))
+        norm = None
+        watermark = None
+
+    # add the images in the top left corner
+    ax = fig.add_subplot(spec[0, 0], projection=wcs)
+    ax_text = fig.add_subplot(spec[0, 1])
+    ax_text.axis('off')
+    ax_starlist = fig.add_subplot(spec[1, 0:])
+    ax_starlist.axis('off')
+
+    ax.imshow(im, origin='lower', norm=norm, cmap='gray_r')
+    ax.set_autoscale_on(False)
+    ax.grid(color='white', ls='dotted')
+    ax.set_xlabel(r'$\alpha$ (J2000)', fontsize='large')
+    ax.set_ylabel(r'$\delta$ (J2000)', fontsize='large')
+    obstime = offset_star_kwargs.get("obstime_isoformat", datetime.datetime.utcnow().isoformat())
+    ax.set_title(f'{source_name} Finder ({obstime})',
+                 fontsize='large', fontweight='bold')
+
+    star_list, _, _, _ = get_nearby_offset_stars(source_ra, source_dec,
+                                                 source_name,
+                                                 **offset_star_kwargs)
+
+    if not isinstance(star_list, list) or len(star_list) == 0:
+        return {'success': False, 'reason': 'failure to get star list'}
```

Comment:
What is happening with this return? Does not seem to conform to docstring.

Reply:
Fixed the docstring and unified the returns.